models/example/order_calculations.sql

```diff
@@ -0,0 +1,30 @@
+with source1 as (
+
+    select * from {{ ref('customers_upload') }}
+
+),
+
+source2 as (
+    
+    select * from {{ ref('orders_upload') }}
+),
+
+calc as (
+
+    select 
+        source1.id as customer_id, 
+        lag(source2.created_at, 1) [ignore nulls] 
+        over (partition by source1.id
+        order by source2.created_at desc) as created_at_prev,
+        created_at
+    from source1
+    
+    inner join source2
+            
+            on source1.id = source2.customer_id
+
+)
+
+    select 
```

Comment:
typically, we create a `final` CTE and the last line of the model is `select * from final` (or whatever the last CTE is). this allows you to easily copy the compiled data to SQL workbench and troubleshoot by running the query and just change which CTE you are querying from (this will likely make more sense if we pair on this tomorrow!)